### ISSUE TITLE
Missing Queue#ready() typescript definition

### ISSUE THREAD

**Issue by myovchev** [bee-queue/bee-queue#172]
Documentation says (https://github.com/bee-queue/bee-queue#ready)

> Instead of listening to this event, consider calling `Queue#ready([cb])`, which returns a Promise that resolves once the Queue is ready. If the Queue is already ready, then the Promise will be already resolved.

There is no matching signature on BeeQueue in the typescript definition

**Comment by myovchev** [comment]
Anyone integrating the library in typescript, a quick (and dirty) solution would be:

```typescript
import BeeQueue, { Job } from 'bee-queue';
import queue from './queue';
import { QueueOptions } from './options';

interface BeeQueheInstance extends BeeQueue {
  ready?(): Promise<any>;
}

export const save = (
  queue: BeeQueheInstance, 
  options: QueueOptions) => async (job: Job) => {
  if (queue.ready) {
    await queue.ready();
  }
  job.data.at = new Date().toISOString();
  // more stuff .......
  return job.save()
}
```

You need optional `ready()` method signature to make it compatible with the current instance definition.

**Comment by myovchev** [comment]
And it seems it is duplicate of #132 and fixed in stale PR  https://github.com/bee-queue/bee-queue/pull/130 more than an year ago.
So my guess is this project is dead?

**Comment by hughsw** [comment]
See PR #187

**Comment by skeggse** [comment]
Duplicate of #132.
